assets/tests/contexts/notificationsContext.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { act as hooksAct, renderHook } from "@testing-library/react-hooks"
+import { mount } from "enzyme"
+import React, { useContext } from "react"
+import { act as testUtilsAct } from "react-dom/test-utils"
+import {
+  NotificationsContext,
+  NotificationsProvider,
+} from "../../src/contexts/notificationsContext"
+import { useNotifications } from "../../src/hooks/useNotifications"
+import { Notification } from "../../src/realtime.d"
+import { now } from "../../src/util/dateTime"
+import { mockUseStateOnce } from "../testHelpers/mockHelpers"
+
+jest.mock("../../src/hooks/useNotifications", () => ({
+  __esModule: true,
+  useNotifications: jest.fn(),
+}))
+
+jest.mock("../../src/laboratoryFeatures", () => ({
+  __esModule: true,
+  default: () => true,
+}))
+
+const notification: Notification = {
+  id: 0,
+  createdAt: now(),
+  reason: "manpower",
+  routeIds: ["route1", "route2"],
+  runIds: ["run1", "run2"],
+  tripIds: [],
+  operatorName: null,
+  operatorId: null,
+  routeIdAtCreation: null,
+}
+
+// tslint:disable: react-hooks-nesting
+
+describe("Notification", () => {
+  test("starts empty", () => {
+    const { result } = renderHook(() => useContext(NotificationsContext), {
+      wrapper: NotificationsProvider,
+    })
+    expect(result.current.notifications).toEqual([])
+  })
+
+  test("receives incoming notifications", () => {
+    let handler: (notification: Notification) => void
+    ;(useNotifications as jest.Mock).mockImplementationOnce((h) => {
```

Comment:
@skyqrose I had been meaning to ask, what's up with this odd `;(useNotifications as jest.Mock)` idiom?

Reply:
The parens are necessary because otherwise it looks like `useNotifciations as (jest.Mock.mockImplementation)`.
The `;` is necessary because otherwise the parens make it look like a function call continued from the previous line: `void(useNotifications)`

(JavaScript is such a disaster that even its syntax has surprises.)